component/src/main/java/me/androidbox/component/event/VisitorSection.kt

```diff
@@ -15,34 +17,94 @@ import androidx.compose.ui.tooling.preview.Preview
+import java.util.UUID
+import kotlin.random.Random
 
 @Composable
 fun VisitorSection(
+    visitors: List<VisitorInfo>,
     modifier: Modifier = Modifier,
-    @StringRes titleResId: Int,
-    content: @Composable () -> Unit
+    visitorDeleteClicked: (VisitorInfo) -> Unit,
+    visitorFilterType: VisitorFilterType = VisitorFilterType.ALL,
 ) {
-    Column(modifier = modifier
-        .fillMaxWidth()) {
+    Column(modifier = modifier) {
+        Spacer(modifier = Modifier.height(16.dp))
 
-        Text(
-            text = stringResource(id = titleResId),
-            fontWeight = FontWeight.Medium,
-            fontSize = 16.sp,
-            color = MaterialTheme.colorScheme.visitorTextFontColor)
+        LazyColumn(modifier = Modifier
+            .fillMaxWidth()
+            .height(300.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp)) {
 
-        Spacer(modifier = Modifier.height(16.dp))
-        content()
+            if (visitors.isNotEmpty()) {
+                /** Going */
+                if (visitorFilterType == VisitorFilterType.ALL || visitorFilterType == VisitorFilterType.GOING) {
+                    item {
+                        Text(
+                            text = stringResource(id = R.string.going),
+                            fontWeight = FontWeight.Medium,
+                            fontSize = 16.sp,
+                            color = MaterialTheme.colorScheme.visitorTextFontColor
+                        )
+                    }
+
+                    items(visitors.filter { visitorInfo -> visitorInfo.isGoing }) { visitor ->
+                        VisitorItem(
+                            visitorInitials = "SM",
+                            visitor = visitor,
+                            isCreator = visitor.isCreator,
+                            onDeleteClicked = visitorDeleteClicked
+                        )
+                    }
+                }
+
+                /* Not going */
+                if (visitorFilterType == VisitorFilterType.ALL || visitorFilterType == VisitorFilterType.NOT_GOING) {
+                    item {
+                        Text(
+                            text = stringResource(id = R.string.not_going),
+                            fontWeight = FontWeight.Medium,
+                            fontSize = 16.sp,
+                            color = MaterialTheme.colorScheme.visitorTextFontColor
+                        )
+                    }
+
+                    items(visitors.filter { visitorInfo -> !visitorInfo.isGoing }) { visitor ->
+                        VisitorItem(
+                            visitorInitials = visitor.initials,
+                            visitor = visitor,
+                            isCreator = visitor.isCreator,
+                            onDeleteClicked = visitorDeleteClicked
+                        )
+                    }
+                }
+            }
+        }
```

Comment:
What I am trying to do here is have `item` which acts like a header for `going` and `not going`
And have 2 `items` blocks that will display the visitors for `going`  or  `not going` or both if `All` is selected
I filter out the what I want display for `going` or `not going` 

`items(visitors.filter { visitorInfo -> visitorInfo.isGoing })`

However, I think it could be inefficient to filter like if this causes multiple  re-composes.